04_numeronym/hsy3418/main.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprintln(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+
+//numeronyms implements a function that returns a slice of numeronyms for its input strings
+func numeronyms(vals ...string) []string {
+
+	numeronymsList := vals
+
+	for index, val := range vals {
+		if len(val) >= 3 {
+			numeronymsList[index] = val[:1] + strconv.Itoa(len(val)-2) + val[len(val)-1:]
```

Comment:
The two ends of the numeronyms are single characters, so using a colon in them makes no sense:
```
numernumsList[index] = val[0] + strconv.Itoa(len(val)-2) + val[len(val)-1]
```